src/libraries/System.Data.OleDb/src/OleDbCommand.cs

```diff
@@ -411,14 +413,38 @@ private void CreateAccessor()
             _dbBindings = bindings;
         }
 
-        private void ApplyParameterBindings(UnsafeNativeMethods.ICommandWithParameters commandWithParameters, tagDBPARAMBINDINFO[] bindInfo)
+        private unsafe void ApplyParameterBindings(UnsafeNativeMethods.ICommandWithParameters commandWithParameters, tagDBPARAMBINDINFO[] bindInfo)
         {
             IntPtr[] ordinals = new IntPtr[bindInfo.Length];
             for (int i = 0; i < ordinals.Length; ++i)
             {
                 ordinals[i] = (IntPtr)(i + 1);
             }
-            OleDbHResult hr = commandWithParameters.SetParameterInfo((IntPtr)bindInfo.Length, ordinals, bindInfo);
+
+            OleDbHResult hr;
+
+            if (s_runningOnX86)
+            {
+                tagDBPARAMBINDINFO_x86[] bindInfo_x86 = new tagDBPARAMBINDINFO_x86[bindInfo.Length];
+                for (int i = 0; i < bindInfo.Length; i++)
+                {
+                    fixed (tagDBPARAMBINDINFO* p = &bindInfo[i])
+                    {
+                        bindInfo_x86[i] = (tagDBPARAMBINDINFO_x86)Marshal.PtrToStructure((IntPtr)p, typeof(tagDBPARAMBINDINFO_x86));
```

Comment:
You can write this as
```
bindInfo_x86[i] = *(tagDBPARAMBINDINFO_x86*)p;
```
It will do the exact same thing and it is going to be like 100x faster.